ulib/axlibc/src/rand.rs

```diff
@@ -6,34 +6,126 @@
  *   THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
  *   See the Mulan PSL v2 for more details.
  */
+use core::ffi::{c_int, c_long, c_uint};
 
-//! Random number generator.
+use core::sync::atomic::{AtomicU64, Ordering::SeqCst};
 
-use core::{
-    ffi::{c_int, c_long, c_uint},
-    sync::atomic::{AtomicU64, Ordering::SeqCst},
-};
+static SEED: AtomicU64 = AtomicU64::new(1);
 
-static SEED: AtomicU64 = AtomicU64::new(0xa2ce_a2ce);
+/// Returns a 32-bit unsigned pseudo random interger from LCG.
+fn random_lcg_32() -> u32 {
+    let new_seed = SEED.load(SeqCst).wrapping_mul(25214903917).wrapping_add(11);
+    SEED.store(new_seed, SeqCst);
+    (new_seed >> 33) as u32
+}
+
+/// Returns a 64-bit unsigned pseudo random interger from LCG.
+fn random_lcg_64() -> u64 {
+    let new_seed = SEED.load(SeqCst).wrapping_mul(6364136223846793005) + 1442695040888963407;
+    SEED.store(new_seed, SeqCst);
+    new_seed as u64
```

Comment:
no need to use `as u64`

Reply:
The warning has been fixed in the lastest commit.